src/io/hdf/hdf_archive.cpp

```diff
@@ -149,6 +151,7 @@ bool hdf_archive::create(const std::filesystem::path& fname, unsigned flags)
 
 bool hdf_archive::open(const std::filesystem::path& fname, unsigned flags)
 {
```

Comment:
Similar to `create()`. make it on the restrictive side.
```
bool hdf_archive::open(const std::filesystem::path& fname, unsigned flags, bool allowed_to_fail = false)
```
This default behavior change may require updating a few test. We can temporarily choose default true and then flip to true in a separate PR. Trying not to grow the size of this PR. Sorry about dealing the legacy code.